nextcord/guild.py

```diff
@@ -2966,3 +2996,161 @@ async def change_voice_state(
+        """|coro|
+
+        Fetch a scheduled event object.
+
+        .. note::
+
+            This is an api call, if updated users is not needed, 
+            consisder :meth:`~Guild.get_scheduled_event`
+
+        Parameters
+        ----------
+        event_id: :class:`int`
+            The event id to fetch
+        with_users: :class:`bool`
+            If the users should be received and cached too, by default False
+
+        Returns
+        -------
+        :class:`ScheduledEvent`
+            The received event object
+        """
+        return await self._state.http.get_event(
+            self.id,
+            event_id,
+            with_users=with_users
+        )
+
+    async def create_scheduled_event(
+        self,
+        *,
+        name: str,
+        entity_type: ScheduledEventEntityType,
+        start_time: datetime.datetime,
+        channel: abc.GuildChannel = MISSING,
+        metadata: EntityMetadata = MISSING,
+        privacy_level: ScheduledEventPrivacyLevel = ScheduledEventPrivacyLevel.guild_only,
+        end_time: datetime.datetime = MISSING,
+        description: str = MISSING,
+    ) -> ScheduledEvent:
+        """|coro|
+
+        Create a new scheduled event object.
+
+        Parameters
+        ----------
+        channel: :class:`abc.GuildChannel`
+            The channel the event will happen in, if any
+        metadata: :class:`EntityMetadata`
+            The metadata for the event
+        name: :class:`str`
+            The name of the event
+        privacy_level: :class:`ScheduledEventPrivacyLevel`
+            The privacy level for the event
+        start_time: :class:`py:datetime.datetime`
+            The scheduled start time
+        end_time: :class:`py:datetime.datetime`
+            The scheduled end time
+        description: :class:`str`
+            The description for the event
+        entity_type: :class:`ScheduledEventEntityType`
+            The type of event
+
+        Returns
+        -------
+        :class:`ScheduledEvent`
+            The created event object.
+        """
+        payload: Dict[str, Any] = {}
+        if channel is not MISSING:
+            payload['channel_id'] = channel.id
+        if metadata is not MISSING:
+            payload['entity_metadata'] = metadata.__dict__
+        if name is not MISSING:
+            payload['name'] = name
+        if privacy_level is not MISSING:
+            payload['privacy_level'] = privacy_level.value
+        if start_time is not MISSING:
+            payload['scheduled_start_time'] = start_time.isoformat()
+        if end_time is not MISSING:
+            payload['scheduled_end_time'] = end_time.isoformat()
+        if description is not MISSING:
+            payload['description'] = description
+        if entity_type is not MISSING:
+            payload['entity_type'] = entity_type.value
+        data = await self._state.http.create_event(self.id, **payload)
+        return self._store_scheduled_event(data)
```

Comment:
The code denotes that most of these as optional, why not document them as such?

Reply:
forgot to update that